### PR TITLE
feat: restrict card donation for users with little progress

### LINKED ISSUE #51293: POST /donate/charge-stripe-card
https://github.com/freeCodeCamp/freeCodeCamp/pull/51598 needs to be added to the charge stripe card endpoint.

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51293
Ref https://github.com/freeCodeCamp/freeCodeCamp/pull/51598

### CODE REVIEW

File: api/src/routes/donate.ts [hunk 1]
```diff
@@ -103,6 +103,15 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
         });
 
         const { email, name } = user;
+        const threeChallengesCompleted = user.completedChallenges.length >= 3;
```
Comment (naomi-lgbt) [inline api/src/routes/donate.ts:106]: Just a thought...

Do we want to require a certification project? Three challenges isn't very many, especially with the granular step-based projects.

Comment (Outlawbeauty) [reply api/src/routes/donate.ts:106]: Hi

**Review by naomi-lgbt — APPROVED** [review]

File: api/src/routes/donate.test.ts [hunk 2]
```diff
@@ -133,6 +173,17 @@ describe('Donate', () => {
...
+
+      it('should return 400 if user has not completed challenges', async () => {
+        await fastifyTestInstance.prisma.user.updateMany({
+          where: { email: userWithProgress.email },
+          data: userWithoutProgress
+        });
+        const failResponse = await superPost('/donate/charge-stripe-card').send(
+          chargeStripeCardReqBody
+        );
+        expect(failResponse.status).toBe(400);
```
Comment (ojeytonwilliams) [inline api/src/routes/donate.test.ts:185]: Could you add the expected response body, too? It's nice to assert the status code, but the thing we really care about is the body.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

File: api/src/routes/donate.test.ts (hunk 2 above)
Comment (ahmaxed) [reply api/src/routes/donate.test.ts:191]: Thanks for the feedback. I included the body in the test accrodingly.

**Review by ojeytonwilliams — APPROVED** [review]
Thanks @ahmaxed! LGTM 👍

**Pull request merged**